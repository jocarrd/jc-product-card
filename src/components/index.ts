import { ProductCardHOCProps } from "../interfaces/interfaces";
import { ProductButtons } from "./ProductButtons";
import { PoductCard as ProductCardHOC } from "./ProductCard";
import { ProductImage } from "./ProductImage";
import { ProductTitle } from "./ProductTitle";

export{ ProductButtons } from "./ProductButtons";
export { ProductTitle } from "./ProductTitle";
export { ProductImage } from "./ProductImage";


export const ProductCard : ProductCardHOCProps = Object.assign(ProductCardHOC,{
    Title:ProductTitle,
    Image : ProductImage,
    Buttons: ProductButtons
})

