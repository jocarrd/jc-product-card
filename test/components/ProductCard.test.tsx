import React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard } from '../../src/components';
import { product2 } from '../data/products';

describe('ProductImage', () => {
  test('ProductCard must be render ', () => {
    const component = renderer.create(
      <ProductCard product={product2}>{() => <h1>ProductCard</h1>}</ProductCard>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
