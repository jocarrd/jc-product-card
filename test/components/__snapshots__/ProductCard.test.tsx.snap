// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage ProductCard must be render  1`] = `
<div
  className="productCard undefined"
>
  <h1>
    ProductCard
  </h1>
</div>
`;

exports[`ProductImage the counter should be incremented 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    ProductCard
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
